### ISSUE TITLE
Editor theme lost after upgrade to RStudio 2022.12.0+353 "Elsbeth Geranium" and cannot be added anymore

### ISSUE THREAD

**Issue by FrancoisR95** [rstudio/rstudio#12650]
### System details

    RStudio Edition : Desktop
    RStudio Version : RStudio 2022.12.0+353 "Elsbeth Geranium"
    OS Version      : Windows 10
    R Version       : 4.1.2 (2021-11-01) -- "Bird Hippie"

### Steps to reproduce the problem

Both parts of the problem can be reproduced independently one from another.
It might be needed to have non-ASCII characters in the name of the Windows user (mine is François) since the error message mentions the path 'C:\Users\François' (although my Rstudio default working directory is 'C:/R-home').

**(1) First part of the problem:**
In a version of RStudio older than 2022.12.0+353 "Elsbeth Geranium", 
in "Tools->Global options->Appearance", add an editor theme and select it for use.
[Subway_ Paris6.rstheme.txt](https://github.com/rstudio/rstudio/files/10538254/Subway_.Paris6.rstheme.txt) (remove the .txt extension in this file for using it)
upgrade to 2022.12.0+353 "Elsbeth Geranium"

=> the editor theme is lost

[EDIT: I could not reproduce this when downgrading and re-upgrading, see details in my comment [https://github.com/rstudio/rstudio/issues/12650#issuecomment-1409516025](url) ]

**(2) Second part of the problem:**
In RStudio 2022.12.0+353 "Elsbeth Geranium"
- in "Tools->Global options->Appearance", try to add an editor theme.
[Subway_ Paris6.rstheme.txt](https://github.com/rstudio/rstudio/files/10538254/Subway_.Paris6.rstheme.txt) (remove the .txt extension in this file for using it)
=> an error message appears
![Bug Rstudio](https://user-images.githubusercontent.com/66539361/215547654-1b0125eb-1921-497d-be7a-ccf070bec19c.png)

[EDIT:
I could reproduce this problem in all of the following RStudio versions:
- 2022.12.0+353 "Elsbeth Geranium"
- 2022.07.2+576.pro12 "Spotted Wakerobin"
- 2021.09.1-372
- 1.3.1056
- 1.3.959

The following version is however not affected by this problem
- 1.2.5042

See details in my comment [https://github.com/rstudio/rstudio/issues/12650#issuecomment-1409516025](url) ]

[EDIT2: finally, upgrading R from 4.1.2 to 4.2.2 solved this problem!]

### Describe the problem in detail
**(1) First part of the problem:**
I had created a customized editor theme (Subway_Paris6.rstheme) about 3 years ago, and since then upgraded several times RStudio without any problems.
[Subway_ Paris6.rstheme.txt](https://github.com/rstudio/rstudio/files/10538254/Subway_.Paris6.rstheme.txt) (remove the .txt extension in this file for using it)

However after the latest upgrade of RStudio 
 from version 2022.07.2+576.pro12 "Spotted Wakerobin"
 to version 2022.12.0+353 "Elsbeth Geranium"
the editor theme disappeared and the R code has therefore got no coloring at all.

**(2) Second part of the problem:**
In addition, my cutomized editor theme does not show up anymore in the list in "Tools->Global options->Appearance".
In addition, when I try to add it back there, an error message appears:
![Bug Rstudio](https://user-images.githubusercontent.com/66539361/215547654-1b0125eb-1921-497d-be7a-ccf070bec19c.png)
" _Failed to add theme
Unable to add the theme: 'C:/R-home/.R/rstudio/themes/Subway_Paris6.rstheme'.
The following error occured: Please check file system permissions.
Also, warnings:
   cannot create dir 'C:\Users\François', reason 'Permission denied'_ "

(the directory 'C:\Users\François' already exists, so it's not clear why RStudio says it tried to create it, and not clear either why it even needs it since my Rstudio default working directory is 'C:/R-home'
Note that there is an open bug which relates to non-ASCII characters in a path so this one might be related, or not, to it : #12448 )

[EDIT: this problem has been introduced between versions RStudio-1.2.5042 and RStudio-1.3.959, see details in my comment [https://github.com/rstudio/rstudio/issues/12650#issuecomment-1409516025](url) ]
[EDIT2: finally, upgrading R from 4.1.2 to 4.2.2 solved this problem!]

### Describe the behavior you expected
I expected my editor theme to remain after the upgrade, and I expected to be able to add it back.

- [x ] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@FrancoisR95 Thank you for raising this! As you've noted, we've run into multiple issues with non-ASCII characters, including with with Windows file paths, and we're often chasing these down. It's possible that the change in the latest release from the Qt framework to Electron is causing some of these to resurface.

@jonvanausdeln Can you try and reproduce the issue on Windows?

**Comment by FrancoisR95** [comment]
Thank you @ronblum 

**Here is a partial workaround and some additional information:**

I downgraded to 2022.07.2+576.pro12 "Spotted Wakerobin" and tried to add back my customized editor theme but unfortunately I still got exactly the same error message.
![Bug RStudio Theme](https://user-images.githubusercontent.com/66539361/215616130-bd3998b4-2dcb-4837-8e12-2e8de622eba2.png)

Then I downgraded to RStudio-2021.09.1-372, then to RStudio-1.3.1056, then to RStudio-1.3.959 (the exact reverse of the upgrade path I had followed in the past).
For each of these versions, I tried to to add the editor theme but still got exactly the same error message.

Then I downgraded to RStudio-1.2.5042, which is the version under which I had initially added the editor theme (in June 2020).
And with that version finally IT WORKED: I could restore my editor theme !
![Bug RStudio Theme - working under RStudio-1 2 5042](https://user-images.githubusercontent.com/66539361/215616240-dbda53da-283b-46cf-834e-928ce9b11f7f.png)
I added also a few other editor themes which had disappeard.
I noticed that this copies the .rstheme files into the 'C:\Users\François\Documents\.R\rstudio\themes' folder.

Then I upgraded directly to 2022.07.2+576.pro12 "Spotted Wakerobin" and my editor themes were still there.
![Bug RStudio Theme - still there under RStudio-2022 07 2-576](https://user-images.githubusercontent.com/66539361/215618595-2fe2b4b4-835f-4d24-8fcc-2b82e046cb73.png)

Then I upgraded to RStudio 2022.12.0+353 "Elsbeth Geranium" and my editor themes were still there.
![Bug RStudio Theme - still there under RStudio-2022 12 0-353](https://user-images.githubusercontent.com/66539361/215618540-6805087d-c7b5-4902-ae4d-7cc44d3bdbce.png)

I don't understand why the themes were removed the first time I upgraded from 2022.07.2+576.pro12 to 2022.12.0+353 but not the second time (maybe because I had initially downgraded? or maybe because I skipped intermediate versions when upgrading from 1.2.5042 to 2022.12.0+353?)

Anyway, this downgrade and upgrade process was finally a successfull workaround.
**However I still cannot add new editor themes. So this bug (the second problem) remains and has been introduced between RStudio-1.2.5042 and RStudio-1.3.959.**
